packages/subproviders/src/utils/wallet_utils.ts

```diff
@@ -0,0 +1,88 @@
+import ethUtil = require('ethereumjs-util');
+import HDNode = require('hdkey');
+import * as _ from 'lodash';
+
+import { DerivedHDKey, WalletSubproviderErrors } from '../types';
+
+const DEFAULT_ADDRESS_SEARCH_OFFSET = 0;
+const BATCH_SIZE = 10;
+const DEFAULT_ADDRESS_SEARCH_LIMIT = 1000;
+
+class DerivedHDKeyIterator implements IterableIterator<DerivedHDKey> {
+    private _initialDerivedKey: DerivedHDKey;
+    private _searchLimit: number;
+    private _index: number;
+
+    constructor(initialDerivedKey: DerivedHDKey, searchLimit: number = DEFAULT_ADDRESS_SEARCH_OFFSET) {
```

Comment:
Why is `searchLimit` defaulted to `DEFAULT_ADDRESS_SEARCH_OFFSET`? Aren't those two completely different things?

Reply:
Did you mean `DEFAULT_ADDRESS_SEARCH_LIMIT`?